hotshot-query-service/src/data_source/storage/sql/queries/node.rs

```diff
@@ -335,25 +380,48 @@ impl<Types: NodeType> UpdateAggregatesStorage<Types> for Transaction<Write> {
                     }
                     *height += 1;
 
-                    *tx_count += block.num_transactions;
-                    *size += block.size;
-                    Some(Ok((block.height as i64, *tx_count as i64, *size as i64)))
+                    *tx_count.entry(None).or_insert(0) += block.num_transactions as usize;
+                    *size.entry(None).or_insert(0) += block.size as usize;
+
+                    // None represents the total of all the namespaces
+                    // and is represented as -1 in databasr
+                    rows.push((
+                        block.height as i64,
+                        -1,
+                        tx_count[&None] as i64,
+                        size[&None] as i64,
+                    ));
+
+                    for (&ns_id, info) in &block.namespaces {
+                        let key = Some(ns_id);
+
+                        *tx_count.entry(key).or_insert(0) += info.num_transactions as usize;
+                        *size.entry(key).or_insert(0) += info.size as usize;
+
+                        rows.push((
+                            block.height as i64,
+                            ns_id as i64,
+                            tx_count[&key] as i64,
+                            size[&key] as i64,
+                        ));
+                    }
+
+                    Some(Ok((block.height as i64, tx_count.clone(), size.clone())))
```

Comment:
Is there a way to avoid cloning here? These could be maps with many namespaces so could potentially be a lot of memory usage